core/drivers/crypto/hisilicon/hpre_ecc.c

```diff
@@ -780,8 +819,7 @@ static enum hisi_drv_status hpre_ecc_dh_params_alloc(struct hpre_ecc_msg *msg)
 static void hpre_ecc_request_deinit(struct hpre_ecc_msg *msg)
 {
 	if (msg->key) {
-		memzero_explicit(msg->key, ECC_DH_KEY_SIZE(msg->key_bytes));
-		free(msg->key);
```

Comment:
This change is OK in itself but I am wondering if there are guarantees that `free_wipe()` will always overwrite the content of the buffer. It does a regular `memset()`, it does not use the trick implemented by `memzero_explicit()` (the volatile function pointer). The generated code should be checked.
It would be good to have this change in a separate commit. It is not related to adding ECC sign and verify.

Reply:
OK, I will fix this after this commit done.

Reply:
solved